### ISSUE TITLE
not an issue but a question:  whats the quality of this library?

### ISSUE THREAD

**Issue by hectoralicea** [eiiches/jackson-jq#12]
I'm  using jq-1.5 (the executable version written in C), and am amazed how much REST API programming I can do in bash simply using curl and jq.

But now we are creating complex JSON objects that don't do well when being parsed by a spring boot app.  Its not really feasible to convert JSON objects to Java Value Objects.  We want the Value Objects to be populated fetching attributes values from a JSON file using JQ rather than force the creation of a complex Java object just so that it can be streamed to and from our "Business Object" represented in JSON with embedded objects and such.  We also want to be able to account for additional attributes that are added dynamically to the JSON data structure without breaking the code.

Will this library make it just as easy to pull attributes out of complex JSON structures, or transform JSON structures as it with JQ in a bash shell script?

**Comment by eiiches** [comment]
Yes, I think. I also use jackson-jq for transforming responses from APIs before mapping to Java objects, eliminating the intermediate Java objects.

You should try `jackson-jq` tool and see if the exact jq code you want to use work or not.
https://github.com/eiiches/jackson-jq#using-a-jackson-jq-command-line-tool
Also, the list of features not yet implemented in jackson-jq might be of help.
https://github.com/eiiches/jackson-jq#differences-between-jq-and-jackson-jq

Does this answer your question?

**Comment by marcosscriven** [comment]
Wanted to say firstly thanks a lot for this excellent implementation of jq!

Secondly, I had a similar question - I see you've implemented this from scratch, starting with a javacc parser definition, and implementing nearly everything. An absolutely epic amount of work!

So I wondered if/how you planned to maintain feature parity?

**Comment by eiiches** [comment]
Thank you for your comment!

jackson-jq aims to be a compatible jq-1.5 (or the future version of jq once it's released) implementation.

However, as jackson-jq being a java library, some features may differ when/if implemented in the future: Streaming, I/O, and Modules.

 - For Streaming and I/O feature, I'm not sure if these can be added to jackson-jq. Actually, I have never tried the features on jq myself yet.
 - For Modules, I've been thinking to make it more like a java integration feature which enables importing static methods directly from java classes or something like that.

Other than those, any incompatible behaviors in core language features and built-in functions are considered defects and should be fixed. The biggest issue I'm aware of currently is the lack of [Complex Assignment](https://stedolan.github.io/jq/manual/#Complexassignments) support.

That said, the development will probably be slow. I will continue to work on this, but I haven't been able to spare much time recently.

BTW, I will be releasing jackson-jq v0.0.7 in a few days, which comes with some jq-1.5 features, such as destructuring, regular expressions (with Oniguruma), and so on.

**Comment by larsga** [comment]
FWIW, I did an experimental Java integration this week that enables me to call static Java methods. Basically, it's a `Function` implementation which wraps a `java.lang.reflect.Method` and then takes care of the type conversions. It's experimental, so it only handles some of the types, but the basic approach works. I wrote it in Scala, though.
